aws/resource_aws_iot_certificate.go

```diff
@@ -28,26 +28,58 @@ func resourceAwsIotCertificate() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+			"certificate_pem": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+			"public_key": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+			"private_key": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
 		},
 	}
 }
 
 func resourceAwsIotCertificateCreate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).iotconn
 
-	log.Printf("[DEBUG] Creating certificate from csr")
-	out, err := conn.CreateCertificateFromCsr(&iot.CreateCertificateFromCsrInput{
-		CertificateSigningRequest: aws.String(d.Get("csr").(string)),
-		SetAsActive:               aws.Bool(d.Get("active").(bool)),
-	})
+	if d.Get("csr") != "" {
```

Comment:
Since `d.Get()` returns an interface, we should either check it after type assertion or switch the check to `d.GetOk()` which will automatically check the value against the zero value for the type:

```suggestion
	if _, ok := d.GetOk("csr"); ok {
```